### ISSUE TITLE
An attribute to Show/Hide/Enable/Disable an entire LayoutGroup

### ISSUE THREAD

**Issue by laurentopia** [TylerTemp/SaintsField#73]
I tried a few combination but they don't seem to work

**Comment by TylerTemp** [comment]
No at this point as `ShowIf` is only a `PropertyDrawer` which only works on a single field.

**Comment by laurentopia** [comment]
np

**Comment by TylerTemp** [comment]
Hi,

this feature has been implemented in [3.7.0](https://github.com/TylerTemp/SaintsField/releases/tag/3.7.0)

```csharp
using SaintsField.Playa;

public bool editableMain;
public bool visibleMain;

[LayoutEnableIf(nameof(editableMain))]
[LayoutShowIf(nameof(visibleMain))]
[LayoutStart("Main", ELayout.FoldoutBox)]
public bool editable1;
public bool visible1;

[LayoutEnableIf(nameof(editable1))]
[LayoutShowIf(nameof(visible1))]
[LayoutStart("./1", ELayout.FoldoutBox, marginBottom: 10)]
public int int1;
public string string1;

[LayoutStart("..", ELayout.FoldoutBox)]
public bool editable2;
public bool visible2;


[LayoutEnableIf(nameof(editable2))]
[LayoutShowIf(nameof(visible2))]
[LayoutStart("./2", ELayout.FoldoutBox)]
public int int2;
public string string2;
```

[![video](https://github.com/user-attachments/assets/f437ebe4-b4f0-4d3e-be8b-646dbdb74eca)](https://github.com/user-attachments/assets/fac5fce5-6458-4853-893c-23fa50f84872)

**Comment by laurentopia** [comment]
Fantastic, I just removed a trove of individual HideIf with one for layout. So useful.
I noticed that the lists ident well in your example. Is this because UITK moved the fold triangle inside the block? I suppose proper indentation in IMGUI is still not possible.
![image](https://github.com/user-attachments/assets/cced6316-05de-45db-926a-11b2b73e43f5)

**Comment by TylerTemp** [comment]
Yes, UI Toolkit got this issue fixed in [3.4.11](https://github.com/TylerTemp/SaintsField/releases/tag/3.4.11), which has not been implemented in IMGUI yet. It'll be processed in the feature (but not sure when tho... depending on how much free time I got...)
